tests/cases/conformance/types/mapped/mappedTypeConstraints2.ts

```diff
@@ -46,6 +46,14 @@ function f6<K extends string>(obj: Mapped6<K>, key: keyof Mapped6<K>) {
   let s: `_${string}` = obj[key]; // Error
 }
 
+type Mapped7<K extends string> = {
+  [P in K as [P] extends [`_${string}`] ? P : never]: P;
+};
+
+function f7<K extends string>(obj: Mapped7<K>, key: keyof Mapped7<K>) {
+  let s: `_${string}` = obj[key];
+}

```

Comment:
This test case is an extra thing that gets fixed here. This is based on `Mapped5`+`f5` from this file. The only difference between them is that this one is using `[P] extends [...]` and not `P extends ...`.

This didn't work because the index type deferral was previously dependent on `hasDistributiveNameType`. I don't quite see a reason why those 2 would type check differently though. Effectively `P` is always instantiated with a non-union type.